### ISSUE TITLE
CFFileDescriptor Manager Thread Consumes 90-100% of the CPU on Linux When Idle

### ISSUE THREAD

**Issue by gerickson** [gerickson/opencflite#89]
First reported as [Open HLX issue #15](https://github.com/gerickson/openhlx/issues/15), it was discovered that the `CFFileDescriptor` manager thread consumes 90-100% of the CPU on Linux when idle:

```
  S     0  264130  264130  264129  0  80   0 - 5368742431 - pts/1    00:00:00 hlxproxyd
1 S     0  264130  264144  264129  0  80   0 - 5368742431 - pts/1    00:00:00 libkqueue_mon
1 R     0  264130  264145  264129 99  80   0 - 5368742431 - pts/1    00:03:04 CFFileDescriptor
1 S     0  264130  264146  264129  0  80   0 - 5368742431 - pts/1    00:00:00 CFSocket
```

**Comment by gerickson** [comment]
The root cause of the problem occurs when the all of the descriptors that the manager thread was previously selecting on have invalidated and closed and there are no longer any descriptors to monitor.

Ideally, the manager thread should block on its internal wakeup read pipe descriptor when there are no descriptors to manage and monitor.

**Comment by gerickson** [comment]
The root cause of the "no longer any descriptors" to monitor was documented by #90 and addressed in #91.

The remaining issue is that the timeout value passed to select is always zero rather than NULL, as it is in the peer `CFSocket` implementation. The distinction between zero and NULL is critical. The former effects a poll whereas the later is an indefinite block until a watch becomes ready.

Since the `CFFileDescriptor` `select` timeout is always zero, this effects a busy-waiting poll that consumes all of the CPU, even when there is no work being done.

**Comment by gerickson** [comment]
Closed with #104.
